File: packages/device-pressure-measurement-native-backend/ios/Plugin/MemoryPressurePlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(MemoryPressurePlugin, "MemoryPressure",
           CAP_PLUGIN_METHOD(memoryInfo, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(removeAllListeners, CAPPluginReturnNone);
)

File: packages/device-pressure-measurement-native-backend/ios/Plugin/MemoryPressurePlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(MemoryPressurePlugin)
public class MemoryPressurePlugin: CAPPlugin {
    private let implementation = MemoryPressure()
    public override func load() {
        super.load()
        NotificationCenter.default.addObserver(self, selector: #selector(self.methodOfReceivedNotification(notification:)), name: UIApplication.didReceiveMemoryWarningNotification, object: nil)
    }
    @objc public func memoryInfo(_ call: CAPPluginCall){
        call.resolve(implementation.memoryInfoResult())
    }
    @objc private func methodOfReceivedNotification(notification:Notification){
        if self.hasListeners("lowMemoryWarning") {
            self.notifyListeners("lowMemoryWarning", data: implementation.memoryInfoResult())
        }
    }
}
